src/Server.ts

```diff
@@ -0,0 +1,177 @@
+import { IClient, IMessage, MessageType } from './Interfaces';
+import * as dgram from 'dgram';
+
+export default class Server
+{
+    private _port: number = 8000;
+    private _socket: dgram.Socket;
+    private _clients: {
+        [id: number]: IClient;
+    } = {};
```

Comment:
Pro tip: TypeScript has an utility type called [`Record`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype) which is more readable and compact

Reply:
Thanks for the tip, didn't know about that!